types/joi/index.d.ts

```diff
@@ -1071,7 +1071,8 @@ export function reach<T extends Schema>(schema: ObjectSchema, path: string): T;
 /**
  * Creates a new Joi instance customized with the extension(s) you provide included.
  */
-export function extend(extention: Extension): any;
+export function extend(extention: Extension|Extension[]): any;
```

Comment:
I see a few potential issues with having the `...` overload:

- You don't need the first overload anymore, since the `...extensions` form covers it
- You allow 0-length arguments, which I believe is forbidden by the API. If you want a variadic function with at least one argument, you're better off doing something like `function extend(extention: string|string[], ...rest: (string|string[])[]): any;`;
- All of these combinations will work with your typings -- some of them look a bit odd to me, but perhaps the library allows them?

```ts
extend(extension)
extend([extension])
extend(extension, extension)
// the odd ones
extend()
extend(extension, [extension])
extend([extension], [extension])
```

Reply:
- Removed the overload
- Require at least 1 argument now
- The odd ones *are* odd, but the library appears to do a single-level flatten on `...args`, so they would work. I don't think there's any issues by allowing people to write "odd" code, since it would still be functionally correct. ¯\_(ツ)_/¯

Reply:
Yep agreed. Thanks!